monitoring/instrumented_mutex.cc

```diff
@@ -34,6 +34,22 @@ void InstrumentedMutex::Lock() {
 void InstrumentedMutex::LockInternal() {
 #ifndef NDEBUG
   ThreadStatusUtil::TEST_StateDelay(ThreadStatus::STATE_MUTEX_WAIT);
+#endif
+#ifdef COERCE_CONTEXT_SWITCH
+  if (stats_code_ == DB_MUTEX_WAIT_MICROS) {
+    thread_local Random rnd(301);
+    if (rnd.OneIn(2)) {
+      if (bg_cv_) {
+        bg_cv_->SignalAll();
+      }
+      sched_yield();
```

Comment:
Probably this should be an alternative to sleeping, or removed. If we're going to sleep 10/11 times below I imagine this doesn't make much difference. I wouldn't mind putting the sleep in an else branch to make it happen less frequently

Reply:
Fixed